### PR TITLE
Chore/metadata refactoring 2

### PR DESCRIPTION
**Pull request by mroz22**
## Description

here I continue (#9218) adding changes to make review of #9130 feasible

## Commits

[1c188b0](https://github.com/trezor/trezor-suite/pull/9262/commits/1c188b0631414a6c503a9fad90106a2dc725d2a9) -  it is not needed to sync all the keys every time. once keys are there, they never change. when migrations come to play we call these syncs relatively more often.
[c351707](https://github.com/trezor/trezor-suite/pull/9262/commits/c351707d8e443bcb85246c2dcbbe9741fcd892bf) - moving constatns to constants. they are going to be reused in metadataActions
[3049739](https://github.com/trezor/trezor-suite/pull/9262/commits/3049739f2af1048879d9f2572482505de8043932) - add actions refactor
[3507789](https://github.com/trezor/trezor-suite/pull/9262/commits/3507789de7c5593208e1832fe5244cd450355a41) - unreleated, when runnign this test locally, it wouldn't clear defaults and fail on subsequent runs.

### CODE REVIEW

File: packages/suite/src/actions/suite/metadataActions.ts [hunk 1]
```diff
@@ -909,11 +908,8 @@ export const init =
             if (!providerResult) {
                 dispatch({ type: METADATA.SET_INITIATING, payload: false });
                 dispatch({ type: METADATA.SET_EDITING, payload: undefined });
-
-                return false;
```
Comment (mroz22) [inline packages/suite/src/actions/suite/metadataActions.ts:907]: this probably should not be removed

File: packages/suite/src/actions/suite/constants/metadataConstants.ts [hunk 2]
```diff
@@ -39,3 +43,12 @@ export const GOOGLE_IMPLICIT_FLOW_CLIENT_ID =
 export const DROPBOX_CLIENT_ID = 'wg0yz2pbgjyhoda';
 
 export const ENCRYPTION_VERSION: MetadataEncryptionVersion = 1;
+export const DEFAULT_ACCOUNT_METADATA: AccountLabels = Object.freeze({
+    accountLabel: '',
+    outputLabels: {},
+    addressLabels: {},
+});
+
+export const DEFAULT_WALLET_METADATA: WalletLabels = Object.freeze({
```
Comment (matejkriz) [inline packages/suite/src/actions/suite/constants/metadataConstants.ts]: not sure if Object.freeze is worth the performance overhead here 🤔 especially if you do the `JSON.parse(JSON.stringify())` anyway

Comment (mroz22) [reply packages/suite/src/actions/suite/constants/metadataConstants.ts]: yes,feels redundant.

File: packages/suite/src/actions/suite/metadataActions.ts [hunk 3]
```diff
@@ -568,81 +576,50 @@ export const connectProvider =
...
+
         // todo: not danger overwrite empty?
         const metadata = fileName ? provider.data[fileName] : undefined;
 
         const nextMetadata = metadata
             ? JSON.parse(JSON.stringify(metadata))
-            : {
-                  walletLabel: '',
-              };
+            : JSON.parse(JSON.stringify(METADATA.DEFAULT_WALLET_METADATA));
```
Comment (matejkriz) [inline packages/suite/src/actions/suite/metadataActions.ts]: nitpick - cloneObject from @trezor/utils can be used
```javascript
const nextMetadata = cloneObject(metadata ?? METADATA.DEFAULT_WALLET_METADATA);
  ```

File: packages/suite/src/actions/suite/metadataActions.ts [hunk 4]
```diff
@@ -655,32 +632,30 @@ export const addAccountMetadata =
...
-        const nextMetadata = metadata
-            ? JSON.parse(JSON.stringify(metadata))
-            : {
-                  outputLabels: {},
-                  addressLabels: {},
-                  accountLabel: '',
-              };
+        const nextMetadata = data
+            ? (JSON.parse(JSON.stringify(data)) as AccountLabels)
+            : JSON.parse(JSON.stringify(METADATA.DEFAULT_ACCOUNT_METADATA));
```
Comment (matejkriz) [inline packages/suite/src/actions/suite/metadataActions.ts]: Note to myself - check if `as AccountLabels` can be avoided here easily

Comment (mroz22) [reply packages/suite/src/actions/suite/metadataActions.ts]: probably yes. I am checking it now

File: packages/suite/src/actions/suite/metadataActions.ts (hunk 3 above)
Comment (mroz22) [reply packages/suite/src/actions/suite/metadataActions.ts]: 1e3737026556471e2ef495e04b31489ff66e0662

File: packages/suite/src/actions/suite/metadataActions.ts (hunk 4 above)
Comment (mroz22) [reply packages/suite/src/actions/suite/metadataActions.ts]: uh, not so easily :( it would require a typequard

Comment (matejkriz) [reply packages/suite/src/actions/suite/metadataActions.ts]: maybe it's a good idea to add the sanitization (and typeguard) as a followup 🤔

**Comment by matejkriz** [comment]
@mroz22 prettier check failed, other than that LGTM

File: packages/suite/src/actions/suite/metadataActions.ts (hunk 4 above)
Comment (mroz22) [reply packages/suite/src/actions/suite/metadataActions.ts]: suuuure 🤙

**Review by matejkriz — APPROVED** [review]
looks valid and still works as expected

**Pull request merged**